packages/twenty-front/src/modules/views/view-picker/hooks/useViewPickerPersistView.ts

```diff
@@ -77,11 +85,22 @@ export const useViewPickerPersistView = () => {
           )[0].id,
         );
         await removeView(viewPickerReferenceViewId);
+        const [indexOrFirstView] = viewsOnCurrentObject
```

Comment:
Could this be turned into a util ? getFirstAvailableView(referenceViewId?: string)

Reply:
Won't require anymore, as we now have single source of truth. now we have setters directly in QueryParamsViewIdEffect, will write description once i make all changes. Thanks.